app/photosketchingcontroller.h

```diff
@@ -0,0 +1,102 @@
+/***************************************************************************
+*                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef PHOTODRAWINGCONTROLLER_H
+#define PHOTODRAWINGCONTROLLER_H
+
+#include <QPointF>
+#include <QObject>
+#include <QColor>
+#include <QtQml/qqmlregistration.h>
+
+
+/**
+ * Utility class, which is used in place of std::pair, because of QML limitations.
+ */
+class ColorPath
+{
+    Q_GADGET
+    Q_PROPERTY( QColor color MEMBER mColor )
+    Q_PROPERTY( QVector<QPointF> points MEMBER mPoints )
+  public:
+    ColorPath( const QColor color, const QVector<QPointF> &list ): mColor( color ), mPoints( list ) {};
```

Comment:
This could be const ref, right?
```suggestion
    ColorPath( const QColor &color, const QVector<QPointF> &list ): mColor( color ), mPoints( list ) {};
```